client/my-sites/plans-v2/products-column/index.tsx

```diff
@@ -0,0 +1,122 @@
+/**
+ * External dependencies
+ */
+import React, { useMemo } from 'react';
+import { translate } from 'i18n-calypso';
+import { useSelector } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { durationToText } from '../utils';
+import { PRODUCTS_TYPES } from '../constants';
+import { JETPACK_PRODUCTS_LIST } from 'lib/products-values/constants';
+import {
+	getJetpackProductDescription,
+	getJetpackProductDisplayName,
+	getJetpackProductShortName,
+	getJetpackProductTagline,
+	isMonthly,
+	getProductFromSlug,
+} from 'lib/products-values';
+import { TERM_ANNUALLY } from 'lib/plans/constants';
+import { getProductCost, isProductsListFetching } from 'state/products-list/selectors';
+import { getCurrentUserCurrencyCode } from 'state/current-user/selectors';
+import getSiteProducts from 'state/sites/selectors/get-site-products';
+import JetpackProductCard from 'components/jetpack/card/jetpack-product-card';
+import FormattedHeader from 'components/formatted-header';
+
+/**
+ * Type dependencies
+ */
+import type { Duration, PurchaseCallback, ProductType } from '../types';
+import type { Product } from 'lib/products-values/products-list';
+
+interface ProductsColumnType {
+	duration: Duration;
+	onProductClick: PurchaseCallback;
+	productType: ProductType;
+	siteId: number | null;
+}
+
+type ProductWithBought = Product & { owned?: boolean };
+
+const ProductComponent = ( {
+	product,
+	onClick,
+	currencyCode,
+}: {
+	product: ProductWithBought;
+	onClick: PurchaseCallback;
+	currencyCode: string;
+} ) => {
+	const price = useSelector( ( state ) => getProductCost( state, product.product_slug ) ) || 0;
+	return (
+		<JetpackProductCard
+			iconSlug={ product.product_slug }
+			productName={ getJetpackProductDisplayName( product ) }
+			subheadline={ getJetpackProductTagline( product ) }
+			description={ getJetpackProductDescription( product ) }
+			currencyCode={ currencyCode }
+			billingTimeFrame={ durationToText( product.term ) }
+			buttonLabel={ translate( 'Get %s', {
```

Comment:
:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?